tests/data/test_unsupervised_sampler.py

```diff
@@ -117,3 +117,15 @@ def test_walker_custom(line_graph):
     for context_pairs, labels in batches:
         for node, neighbour in context_pairs[labels == 1]:
             assert node == neighbour
+
+
+def test_ignored_param_warning(line_graph):
+    walker = UniformRandomWalk(line_graph, n=2, length=3)
+    with pytest.warns(UserWarning):
```

Comment:
```suggestion
    with pytest.warns(UserWarning, match="walker, length"):
```
or going all out
```suggestion
    with pytest.warns(UserWarning, match="walker, length: 'walker' was provided so 'length' parameter will be ignored"):
```

and so on for the other ones.

I wonder if makes sense to just make these `ValueError`s too, instead of warnings; it means slightly fewer code paths for us to think about and less risk of mixing something up (e.g. we might refactor `UnsupervisedSampler` to use `seed` in the `__init__` unconditionally and a warning wouldn't necessarily be as obvious to us), but doesn't really matter that much. Thoughts?

Reply:
Hmm that's true, yeah I can update it to be `ValueError` instead. It doesn't quite catch everything though since:
```python
sampler = UnsupervisedSampler(g, seed=None, walker=some_walker)
```
won't raise an error if the user intended to use no seed, since `None` is the default value. 

Maybe we should consider just deprecating the other parameters and use the `walker` parameter as the only way to specify additional parameters - for instance:
```python
# use defaults
sampler = UnsupervisedSampler(g)

# use a seed
sampler = UnsupervisedSampler(g, walker=UniformRandomWalk(g, seed=0))
```